src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -288,20 +302,256 @@ private static ProjectItemGroupElement CreateItemGroup(Project project, string f
+
+                    result.Add(tfm.ToString(), tfm.Dependencies.Select(d => {
+
+                        
+                        string versionStr; //requested version exactly as it is in the proj file
+                        VersionRange version; //requested version value as a range
+                        //If the package is not auto-referenced, get the version from the project file. Otherwise fall back on the assets file
+                        if (!d.AutoReferenced)
+                        {
+                            versionStr = projPackages.Where(p => p.EvaluatedInclude.Equals(d.Name)).First().GetMetadataValue(VERSION_TAG);
+                            version = VersionRange.Parse(versionStr);
+                        }
+                        else
+                        {
+                            versionStr = d.LibraryRange.VersionRange.ToString();
+                            version = d.LibraryRange.VersionRange;
+                        }
+
+                        return new PRPackage {
+                            name = d.Name,
+                            requestedVersion = version,
+                            projectFileRequestedVersion = versionStr,
+                            autoReference = d.AutoReferenced
+                        };
+                    }));
+
+                }
+
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Prepares the dictionary that maps frameworks to packages top-level
+        /// and transitive.
+        /// </summary>
+        /// <param name="userInputFrameworks">A list of framework names</param>
+        /// <param name="assetsFile">Assets file for all targets and libraries</param>
+        /// <param name="requestedVersions">Top-level packages with the requested versions</param>
+        /// <param name="transitive">Include transitive packages in the result</param>
+        /// <returns></returns>
+        private Dictionary<string, Tuple<IEnumerable<PRPackage>, IEnumerable<PRPackage>>> GetResolvedVersions(
+            IList<string> userInputFrameworks,LockFile assetsFile,
+            Dictionary<string, IEnumerable<PRPackage>> requestedVersions, bool transitive)
+        {
+            var resultPackages = new Dictionary<string, Tuple<IEnumerable<PRPackage>, IEnumerable<PRPackage>>>();
+
+            //Gets the targets from the assets file given the list of frameworks
+            var inputLockFileTargets = new List<LockFileTarget>();
+            foreach (var framework in assetsFile.PackageSpec.TargetFrameworks)
+            {
+                if (userInputFrameworks.Count == 0 || userInputFrameworks.Contains(framework.ToString()))
+                {
+                    inputLockFileTargets.Add(assetsFile.GetTarget(framework.FrameworkName, null));
+                }
+
+            }
+             
+            foreach (var target in assetsFile.Targets)
```

Comment:
You can combine the code to get requested version and resolved version instead of being in separate APIs and running the same loops twice... 

Ideal flow should be you filter our all the input frameworks, get their dependencies and requested versions from framework dependencies in assets file, and then get their resolved versions from target frameworks. Talk to Rohit, he'd be able to help you with this,

Reply:
- The assets file will not always have the same string for version as the project file. It could be confusing to have different version showing in the requested than the one in the project file.

- I can still put them in the same loop, but I thought this might be more readable